events.go

```diff
@@ -0,0 +1,132 @@
+package dht
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	kbucket "github.com/libp2p/go-libp2p-kbucket"
+)
+
+// AsyncEvent is emitted for every notable event that happens during a DHT async lookup.
+// AsyncEvent supports JSON marshalling because all of its fields do, recursively.
+type AsyncEvent struct {
```

Comment:
Async is sort of a weird name, how about Lookup?

Reply:
Done.